dspy/predict/react.py

```diff
@@ -64,7 +64,7 @@ def act(self, output, hop):
 
             if action_name == 'Finish': return action_val
 
-            output[f"Observation_{hop+1}"] = self.tools[action_name](action_val).passages
+            output[f"Observation_{hop+1}"] = self.tools[action_name](action_val)
```

Comment:
@dannyoo I think this would be a sufficient change. 

```suggestion
            try: 
                output[f"Observation_{hop+1}"] = self.tools[action_name](action_val).passages
            except AttributeError:
                # Handle the case where 'passages' attribute is missing
                # TODO: This is a hacky way to handle this. Need to fix this.
                output[f"Observation_{hop+1}"] = self.tools[action_name](action_val)
```

I could open a new PR with this change and bring in your commit, but since it's small I'm just adding this suggestion.